spanner/spanner_snippets/spanner/spanner_create_backup.go

```diff
@@ -40,9 +41,24 @@ func createBackup(w io.Writer, db, backupID string) error {
 		return fmt.Errorf("createBackup.NewDatabaseAdminClient: %v", err)
 	}
 	defer adminClient.Close()
-	dbMetadata, err := adminClient.GetDatabase(ctx, &adminpb.GetDatabaseRequest{Name: db})
+	client, err := spanner.NewClient(ctx, db)
 	if err != nil {
-		return fmt.Errorf("createBackup.GetDatabase: %v", err)
+		return fmt.Errorf("createBackup.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	stmt := spanner.Statement{
```

Comment:
Would it be better (and at all possible) to move this code away from the sample and into the test code. I worry a little bit that this will confuse users a little, as we are showing an example of something that is not really best practice. This is creating a backup using the current timestamp, which is basically the same as creating a backup without setting a version time. Setting a version time is something that a user should normally only do if they want a specific version to be backed up.
In the Java sample we keep the version time selection in the test code and pass that into the test.

Another possibility could be do show an example where we execute a read/write transaction and then backup the database at the commit timestamp of the read/write transaction. That is a somewhat more reasonable use case, as it shows users how to create a backup containing everything up to a specific commit.

Reply:
That is fair, we are looking to change the samples at the moment to something more realistic than the `earliest_version_time`, since the database is not 1 hour old. We will need to revisit this samples after launch, but for now this change would be better than the current version time that we are setting.

Reply:
I agree with @olavloite. I don't think we should be putting this logic in the sample. If instead, like in the C# sample, we can compute the version time in the test and pass it in, that would be ideal.

Reply:
Ok, changed the sample to have the version time parameterised.